tobac/utils/general.py

```diff
@@ -539,21 +535,13 @@ def combine_tobac_feats(list_of_feats, preserve_old_feat_nums=None):
     combined_df = pd.concat(list_of_feats)
     # Then, sort by time first, then by feature number
     combined_df = combined_df.sort_values(["time", "feature"])
-    all_times = sorted(combined_df["time"].unique())
-    # Loop through current times
-    start_feat_num = combined_df["feature"].min()
     # Save the old feature numbers if requested.
     if preserve_old_feat_nums is not None:
         combined_df[preserve_old_feat_nums] = combined_df["feature"]
 
-    for frame_num, curr_time in enumerate(all_times):
-        # renumber the frame number
-        combined_df.loc[combined_df["time"] == curr_time, "frame"] = frame_num
-        # renumber the features
-        curr_row_count = len(combined_df.loc[combined_df["time"] == curr_time])
-        feat_num_arr = np.arange(start_feat_num, start_feat_num + curr_row_count)
-        combined_df.loc[combined_df["time"] == curr_time, "feature"] = feat_num_arr
-        start_feat_num = np.max(feat_num_arr) + 1
-
-    combined_df = combined_df.reset_index(drop=True)
-    return combined_df
+    # count_per_time = combined_feats.groupby('time')['index'].count()
+    combined_df["frame"] = combined_df["time"].rank(method="dense").astype(int) - 1
+    combined_sorted = combined_df.sort_values(["frame", "idx"], ignore_index=True)
```

Comment:
**question:** To clarify, idx is assigned as the index of each feature detected per frame in feature detection, and may not be contiguous due to lower threshold features being removed?

**suggestion:** Is sorting by idx necessary? I guess it's to try and keep features on the same frame in the same order as in the pre-combined dataframes?

Reply:
Yes, `idx` is assigned as the index of each feature detected per frame in feature detection. 

Yeah, I wanted to preserve as much as possible as the original dataframes. I could see users getting confused if sorting on `feature` and sorting on (`frame`, `idx`) produced different results.